UserApi/src/main/java/com/qa/service/UserService.java

```diff
@@ -74,4 +74,48 @@ public String updateUser(Long id, User user) {
 		return repo.findAll();
 	}
 
+	@Override
+	public String tagUser(Long uId, Long aId) {
+		Optional<User> user = repo.findById(uId);
+        String tags = user.get().getTags();
```

Comment:
this check for tags could be a helper method, there's too many things now happening in your method